### PR TITLE
Fix https://github.com/nrc/rustfmt/issues/376

### PR DESCRIPTION
**Pull request by sinhpham**

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -42,7 +42,12 @@ pub fn rewrite_path(context: &RewriteContext,
 
     if let Some(ref qself) = qself {
         result.push('<');
-        result.push_str(&pprust::ty_to_string(&qself.ty));
+        let fmt_ty = if let Some(str) = qself.ty.rewrite(context, width, offset) {
```
Comment (marcusklaas) [inline src/types.rs]: I think it's better to simply return `None` when the type rewrite fails. We'd going over the column limit, otherwise.

**Comment by nrc** [comment]
Could you add a test please?

File: src/types.rs (hunk 1 above)
Comment (sinhpham) [inline src/types.rs]: We're appending the fmt_ty to result, so shouldn't we at least push the original string in case we can't format this?

**Comment by sinhpham** [comment]
Some tests already exist in tests/target/paths.rs, this PR is just a refactor to use ast::Ty::rewrite.

File: src/types.rs (hunk 1 above)
Comment (marcusklaas) [inline src/types.rs]: Not at this level. We can just abandon formatting the path entirely. At some point in the call stack we will push the original string, if we can really find no other way to format things.

Comment (sinhpham) [inline src/types.rs]: Got it, changed to use try_opt!

File: src/types.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 use syntax::ast;
 use syntax::print::pprust;
-use syntax::codemap::{self, Span, BytePos, CodeMap};
+use syntax::codemap::{self, Span, BytePos, CodeMap, mk_sp};
```
Comment (marcusklaas) [inline src/types.rs]: Could you remove the `mk_sp` import? It's not used anymore.

**Comment by marcusklaas** [comment]
Thanks! Except for the tiny nit, this is ready to merge.

File: src/types.rs (hunk 2 above)
Comment (sinhpham) [inline src/types.rs]: Thanks @marcusklaas , I used this for debugging but forgot to remove. PR updated.

**Pull request merged**